arangod/Graph/Providers/BaseProviderOptions.h

```diff
@@ -56,7 +56,9 @@ struct IndexAccessor {
   aql::AstNode* _indexCondition;
   std::optional<size_t> _memberToUpdate;
 
-  // TODO: This needs to be changed BEFORE merge
+  // Note: We would prefer to have this a unique_ptr here.
+  // However the IndexAccessor is used in std::vector<IndexAccessor>
+  // which then refuses to compile (deleted copy constructor)
   std::shared_ptr<arangodb::aql::Expression> _expression;
```

Comment:
Just curious, at what operation did the compilation fail? Generally for resizing of a vector, a move constructor should suffice (and is also preferred if it is `noexcept`).

Reply:
I can show you offline.
If we can sort this out to use a Unique_PTR I am very happy to fix this.
We do NOT share anything here.

Reply:
Thanks for your help!
Figured this out